simulator.py

```diff
@@ -0,0 +1,172 @@
+"""
+Create a simulator to calculate pnl
+"""
+
+import collections
+import copy
+import uuid
+from thousandaire.constants import TRADING_CONFIGS
+from thousandaire.dataset import Data
+
+DATASET = 'price_dataset'
+COST_FUNC = 'cost_function'
+
+class SimulateMethod():
+    """
+    The method for simulating.
+    """
+    def __init__(self, region, instruments):
+        self.quantity = collections.namedtuple('value', ['value', 'quantity'])
```

Comment:
Is there any reason that we need to keep both values (value, quantity) here?
We already pass `price` to cost functions. The function can access price yesterday and price today to get enough information. So I think instead of having a new type here, we can use Portfolio directly.